Fix cookie check cookie made into a session cookie in IE8

shane-tomlinson opened the pull request:
IE8 doesn't know how to handle max-age when setting cookies turning the "cookie check cookie" into a session cookie.  Using expires=<date> as well.  Manually expire the cookie after its existence is checked for.

1982

Commit:
Fix cookie check in IE8

IE8 doesn't know how to handle max-age when setting cookies turning the "cookie check cookie" into a session cookie.  Using expires=<date> as well.  Manually expire the cookie after its existence is checked for.

1982

seanmonstar commented on resources/static/common/js/network.js:
```diff
@@ -715,14 +715,32 @@ BrowserID.Network = (function() {
+          expiryDate.setTime(expiryDate.getTime() + (maxAgeSeconds * 1000));
+
+          document.cookie = "__cookiesEnabledCheck=1; max-age="
+            + maxAgeSeconds + "; expires=" + expiryDate.toGMTString();
```
why set expires in the future? We can just set the cookie, if it can be read, `enabled = true`, and then set expires to `-1` (or anything older than right now).

shane-tomlinson replied:
Good call @seanmonstar on the future expiration.  an expires of -1 turns the cookie into a session cookie.  To delete the cookie, I set the cookie to expire in the past.

Commit:
Simplify the cookie check.

Never set a future expiration date, to delete the cookie set an expiration date of yesterday.

lloyd commented:
@seanmonstar can haz r+ and green button?

seanmonstar commented:
Nicely simplified.
